cpp/community_detection_module/algorithm_online/community_detection.hpp

```diff
@@ -55,62 +57,188 @@ class LabelRankT {
+    }
+
+    return neighbors;
+  }
+
+  ///@brief Initializes the internal data structures.
+  void InitializeStructures(const mgp::Graph &graph, std::uint64_t node_id) {
+    times_updated[node_id] = 0;
+    std::unordered_map<std::uint64_t, double> node_label_Ps;
 
-  ///@brief Initializes internal data structures.
-  void SetStructures(std::uint64_t node_id);
+    const auto in_neighbors = InNeighborsIDs(graph, node_id);
 
-  ///@brief Removes deleted nodes’ entries from internal data structures.
+    double sum_w_i = w_selfloop;
+    for (const auto node_j_id : in_neighbors) {
+      sum_w_i += GetTotalWeightBetween(graph, node_j_id, node_id);
+    }
+
+    // add self-loop
+    node_label_Ps[node_id] = w_selfloop / sum_w_i;
+
+    // add other relationships
+    for (const auto node_j_id : in_neighbors) {
+      node_label_Ps[node_j_id] += GetTotalWeightBetween(graph, node_j_id, node_id) / sum_w_i;
+    }
+
+    label_Ps[node_id] = std::move(node_label_Ps);
+    sum_w[node_id] = sum_w_i;
+  }
+
+  ///@brief Removes deleted nodes’ entries from the internal data structures.
   ///
   ///@param deleted_nodes_ids -- set of deleted nodes’ IDs
-  void RemoveDeletedNodes(
-      const std::unordered_set<std::uint64_t>& deleted_nodes_ids);
+  void RemoveDeletedNodes(const std::unordered_set<std::uint64_t> &deleted_nodes_ids) {
+    for (const auto node_id : deleted_nodes_ids) {
+      label_Ps.erase(node_id);
+      sum_w.erase(node_id);
+      times_updated.erase(node_id);
+    }
+  }
 
-  ///@brief Retrieves the total weight of edges between two nodes.
-  /// With undirected graphs it considers all edges between the given node pair.
+  ///@brief Retrieves the total weight of relationships between two nodes.
+  /// With undirected graphs it considers all relationships between the given node pair.
   /// On a directed graph, it checks if the source and target nodes match.
   ///
   ///@param from_node_id -- source node’s Memgraph ID
   ///@param to_node_id -- target node’s Memgraph ID
   ///
-  ///@return -- total weight of edges between from_node_id and to_node_id
-  double GetTotalWeightBetween(std::uint64_t from_node_id,
-                               std::uint64_t to_node_id) const;
+  ///@return -- total weight of relationships between from_node_id and to_node_id
+  double GetTotalWeightBetween(const mgp::Graph &graph, std::uint64_t from_node_id, std::uint64_t to_node_id) const {
+    double total_weight = 0;
+
+    for (const auto &out_relationship : graph.GetNodeById(mgp::Id::FromUint(from_node_id)).OutRelationships()) {
+      if (out_relationship.To().Id().AsUint() == to_node_id) {
+        total_weight += GetWeight(graph, out_relationship);
+      }
+    }
+    if (!is_directed) {
+      for (const auto &in_relationship : graph.GetNodeById(mgp::Id::FromUint(from_node_id)).InRelationships()) {
+        if (in_relationship.From().Id().AsUint() == to_node_id) {
+          total_weight += GetWeight(graph, in_relationship);
+        }
```

Comment:
Again, you can restructure this to have one less indent.

Reply:
Done